Content.Server/GameTicking/Rules/NukeopsRuleSystem.cs

```diff
@@ -0,0 +1,163 @@
+    private bool _opsWon;
+
+    public override string Prototype => "Nukeops";
+
+    public override void Initialize()
+    {
+        base.Initialize();
+
+        SubscribeLocalEvent<RoundStartAttemptEvent>(OnStartAttempt);
+        SubscribeLocalEvent<RulePlayerSpawningEvent>(OnPlayersSpawning);
+        SubscribeLocalEvent<MobStateChangedEvent>(OnMobStateChanged);
+        SubscribeLocalEvent<RoundEndTextAppendEvent>(OnRoundEndText);
+        SubscribeAllEvent<NukeExplodedEvent>(OnNukeExploded);
+    }
+
+    private void OnNukeExploded(NukeExplodedEvent ev)
+    {
+        _opsWon = true;
+        _roundEndSystem.EndRound();
+    }
+
+    private void OnRoundEndText(RoundEndTextAppendEvent ev)
+    {
+        ev.AddLine(_opsWon ? Loc.GetString("nukeops-ops-won") : Loc.GetString("nukeops-crew-won"));
+        ev.AddLine(Loc.GetString("nukeops-list-start"));
+        foreach (var nukeop in _aliveNukeops)
+        {
+            ev.AddLine($"- {nukeop.Key.Session?.Name}");
+        }
+    }
+
+    private void OnMobStateChanged(MobStateChangedEvent ev)
+    {
+        if (!_aliveNukeops.TryFirstOrNull(x => x.Key.OwnedEntity == ev.Entity, out var op)) return;
+
+        _aliveNukeops[op.Value.Key] = op.Value.Key.CharacterDeadIC;
+
+        if (_aliveNukeops.Values.All(x => !x))
+        {
+            _roundEndSystem.EndRound();
+        }
+    }
+
+    private void OnPlayersSpawning(RulePlayerSpawningEvent ev)
+    {
+        _aliveNukeops.Clear();
+
+        var numOps = (int)Math.Min(Math.Floor((double)ev.PlayerPool.Count / _cfg.GetCVar(CCVars.NukeopsPlayersPerOp)),
+            _cfg.GetCVar(CCVars.NukeopsMaxOps));
+        var ops = new IPlayerSession[numOps];
+        for (var i = 0; i < numOps; i++)
+        {
+            ops[i] = _random.PickAndTake(ev.PlayerPool);
+        }
+
+        var map = "/Maps/syndiepuddle.yml";
+
+        var aabbs = _stationSystem.Stations.SelectMany(x =>
+            Comp<StationDataComponent>(x).Grids.Select(x => _mapManager.GetGridComp(x).Grid.WorldBounds)).ToArray();
+        var aabb = aabbs[0];
+        for (int i = 1; i < aabbs.Length; i++)
+        {
+            aabb.Union(aabbs[i]);
+        }
+
+        var (_, grid) = _mapLoader.LoadBlueprint(GameTicker.DefaultMap, map, new MapLoadOptions
+        {
+            Offset = aabb.Center + Math.Max(aabb.Height, aabb.Width)/2 + new Vector2(-500, -500)
+        });
+
+        if (!grid.HasValue)
+        {
+            Logger.ErrorS("NUKEOPS", $"Gridid was null when loading \"{map}\", aborting.");
+            foreach (var session in ops)
+            {
+                ev.PlayerPool.Add(session);
+            }
+            return;
+        }
+
+        var gear = _prototypeManager.Index<StartingGearPrototype>("SyndicateOperativeGearFull");
+        var spawnpos = new EntityCoordinates(_mapManager.GetGridEuid(grid.Value), new Vector2(1.5f, -4.5f));
+        for (var i = 0; i < ops.Length; i++)
+        {
```

Comment:
Should really use spawn points.

Reply:
yes, however: i want a minimal viable product. imo this should do for now